src/sentry/api/serializers/models/rule.py

```diff
@@ -13,6 +13,22 @@
 from sentry.models.rulefirehistory import RuleFireHistory
 from sentry.models.rulesnooze import RuleSnooze
 from sentry.services.hybrid_cloud.user.service import user_service
+from sentry.utils.request_cache import request_cache
+
+
+@request_cache
+def get_neglected_rule(obj):
+    return NeglectedRule.objects.get(
+        rule=obj,
+        organization=obj.project.organization_id,
+        opted_out=False,
+        sent_initial_email_date__isnull=False,
+    )
+
+
+@request_cache
+def get_rule_snooze(obj, user_id):
```

Comment:
@ceorourke this won't cache well because it's going to cache it per arguments. It needs to have the same arguments for each time we call it. Same with the caching above.

Reply:
Yeah, we should fetch this in bulk in `get_attrs`

Reply:
Or we can cache a single query and filter out the outputs in memory to find the right item. Then you don't need to do it in `get_attrs`.

Reply:
This will still make a query per `RuleSnooze`.

Is the problem here that the rules endpoint has an N + 1? To fix that you don't really need to use caching. Just use get_attrs and Make a query like `RuleSnooze.objects.filter(Q(user_id=user_id) | Q(user_id=None), rule__in=[all_rules])`, and do the same for `NeglectedRule`

Reply:
That works as well